Fixing slow_test store.pruned_load.

clemahieu opened the pull request:
This was crashing on the line store->pruned.del as hashes was empty and querying for .begin (). hashes was never populated even when this test was originally added.

clemahieu committed:
Fixing slow_test store.pruned_load. This was crashing on the line store->pruned.del as hashes was empty and querying for .begin (). hashes was never populated even when this test was originally added.

dsiganos commented on nano/slow_test/node.cpp:
```diff
@@ -443,12 +443,13 @@ TEST (store, pruned_load)
 			{
 				auto transaction (store->tx_begin_write ());
-				for (auto k (0); k < batch_size / 2; ++k)
+				for (auto k (0); !hashes.empty () && k < batch_size / 2; ++k)
```
Do we need the check `!hashes.empty ()`?
This is a unit test, if anything is wrong, we want to fail loud and clear rather than handle it quietly.
So I would say that we should ASSERT that hashes is not empty.

dsiganos committed:
Improve the slow_test store.pruned_load to work the same way for all databases (#3666)

Previously, it did one thing for lmdb and another for rocksdb due to
rocksdb not having an accurate count of blocks.
The block counting is now done manually by iterating through all the blocks.
